Modules/arraymodule.c

```diff
@@ -133,10 +147,12 @@ enum machine_format_code {
 static int
 array_resize(arrayobject *self, Py_ssize_t newsize)
 {
+    _Py_CRITICAL_SECTION_ASSERT_OBJECT_LOCKED(self);
     char *items;
     size_t _new_size;
 
-    if (self->ob_exports > 0 && newsize != Py_SIZE(self)) {
+    if (FT_ATOMIC_LOAD_SSIZE_RELAXED(self->ob_exports) > 0 &&
```

Comment:
I think we always access `ob_exports` while the critical section is held, as such atomic is not needed or am I missing something?

Reply:
`array_buffer_relbuf()`, or I could critical section that as well?

Reply:
Yes, that would avoid us using atomic everywhere else.